### ISSUE TITLE
Suggestion: Make AI build army while attacking

### ISSUE THREAD

**Issue by demoix** [Sh0wdown/UnofficialCrusaderPatch#689]
That would be a great enhancement to every AI character. Currently, AI stops building an army for the next attack while it is on the way to attack someone. There are many AI's that their economy grows faster than he actually has time to spend the money on building an army. Even if they have 60000 gold and keeps growing they do not build any unit while they attacking someone. Can this be changed? Would be great if they keep building up a new army for the second attack if they have plenty of money while the first army is besieging someone. And then when the first army is defeated they could send the second army if they reach enough troops or could still build up until reach a willing number of troops and then send to attack again. And meanwhile, prepare another army for another attack when second failed.

**Comment by GRhin** [comment]
From memory, this was done in a very early version of ucp, well before my time so I might be mistaken. It was removed because it resulted in the AI trickling units out to the existing seige as it recruited them, which also had the effect that the seige was never finished, so they forever trickled troops, never building up for a new attack, essentially breaking the AI

**Comment by GRhin** [comment]
Hmmm. I can find it in the release history, maybe it was just tested and not implemented. Or maybe I'm making it up. @Krarilotus you have been around longer than I, does this feeling a bell?

**Comment by J-T-de** [comment]
Yes, GRhin is right. A feature was tested (but probably not even released, but I also didn't find the link) but scrapped. Instead of collecting a new attack, each single troop was sent to the ongoing siege, and this is not useful. This might be something more involved...
